Zend/zend_alloc.c

```diff
@@ -2261,8 +2269,12 @@ void zend_mm_shutdown(zend_mm_heap *heap, bool full, bool silent)
 	zend_mm_chunk *p;
 	zend_mm_huge_list *list;
 
+	if (full == false) {
+		zend_mm_observers_shutdown(heap);
+	}
```

Comment:
Nit: I suppose this can be in the `#if` to avoid the function call. Not that it really matters.